infrastructure/src/main/java/org/corfudb/infrastructure/management/FailureDetector.java

```diff
@@ -210,7 +234,7 @@ private long modifyIterationTimeouts(Map<String, IClientRouter> router, PollRepo
         long pollIterationStartTime = System.nanoTime();
```

Comment:
Move this pollIterationStartTime to before the start of iteration in the pollRound.
This variable doesn't make sense here.
The `iterationElapsedTime` has to be the time it took to complete 1 poll iteration.

Reply:
I copy pasted this logic from the previous version. Not sure if we need to change it in this PR. Let's create another one especially for timeouts

Reply:
This is definitely a bug. Since you are anyways cleaning this code, you can clean this here.
We do not want to revisit this code again for some time.
Anyways, this is your call.

Reply:
let's leave it for an another pr